driver-async/src/main/com/mongodb/async/client/gridfs/AsyncInputStream.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.async.client.gridfs;
+
+import com.mongodb.async.SingleResultCallback;
+
+import java.nio.ByteBuffer;
+
+/**
+ * The Async Input Stream interface represents some asynchronous input stream of bytes.
+ *
+ * @see {@link com.mongodb.async.client.gridfs.helpers} for adapters to create {@code AsyncInputStream}s.
```

Comment:
Better to link to individual classes rather than a package I think, though I'm not sure since javadoc generation failed for me.